src/test/java/htsjdk/samtools/cram/build/ContainerFactoryTest.java

```diff
@@ -21,105 +19,499 @@
  */
 public class ContainerFactoryTest extends HtsjdkTest {
 
-    @Test
-    public void testUnmapped() {
-        SAMFileHeader header = new SAMFileHeader();
+    public static Container getSingleRefContainer(final SAMFileHeader samFileHeader) {
```

Comment:
these are public because they're reused by ContainerParserTest

Reply:
These get methods are `public static`, and are not annotated as tests, but have asserts. Its a little hard to tell whats test code and whats utility code in here - is it possible to separate them ?

Reply:
Tthere is a lot of code that is duplicated many times in this file (ie., the code that creates `CramCompressionRecord`s) that could be factored out. 

Some of the utility code might be something we can move to CramTestUtils if its useful across tests.